src/tequila/quantumchemistry/orbital_optimizer.py

```diff
@@ -200,8 +235,18 @@ def kernel(self, h1, h2, *args, **kwargs):
                                         **self.molecule_arguments)
         if restrict_to_hcb:
             H = molecule.make_hardcore_boson_hamiltonian()
```

Comment:
there is currently an inconsistency if `use_hcb` and `oo_between_spaces` are both set to `True`

2 Solutions:

either explicitly catch this, and raise an Exception

or adapt your method, that if also works with `use_hcb`. It would go like this then
`X_list = [X(i) for i in self.inactive_spin_orbitals]`
to
`X_list = [X(i) for i in self.inactive_spatial_orbitals]`